includes/class-sensei-analysis.php

```diff
@@ -320,190 +320,124 @@ public function analysis_page() {
-			<div class="sensei-analysis-main">
-				<?php $sensei_analysis_user_course->display(); ?>
-			</div>
-			<div class="sensei-analysis-extra">
-				<?php do_action( 'sensei_analysis_extra' ); ?>
-			</div>
-		</div>
-		<?php
-		do_action( 'analysis_wrapper_container', 'bottom' );
-		do_action( 'analysis_after_container' );
+		$exclude_query_params        = [ 'q' ];
+		$this->display_report_page( $sensei_analysis_user_course, 'user_course', $exclude_query_params );
 	}
 
 	/**
 	 * An individual Course view for analysis, showing all the Learners
 	 *
-	 * @since  1.2.0
-	 * @param integer $course_id
+	 * @since 1.2.0
+	 *
+	 * @param int $course_id
 	 * @return void
 	 */
-	public function analysis_course_users_view( $course_id ) {
-
-		// Load Analysis data
+	public function analysis_course_users_view( int $course_id ) {
 		$sensei_analysis_course_users = $this->load_data_object( 'Course', $course_id );
-		// Wrappers
-		do_action( 'analysis_before_container' );
-		do_action( 'analysis_wrapper_container', 'top' );
-
-		$this->analysis_course_users_nav();
-		do_action( 'sensei_analysis_after_headers' );
-
-		?>
-		<div id="poststuff" class="sensei-analysis-wrap course-profile">
-			<div class="sensei-analysis-main">
-				<?php $sensei_analysis_course_users->display(); ?>
-			</div>
-			<div class="sensei-analysis-extra">
-				<?php do_action( 'sensei_analysis_extra' ); ?>
-			</div>
-		</div>
-		<?php
-		do_action( 'analysis_wrapper_container', 'bottom' );
-		do_action( 'analysis_after_container' );
+		$exclude_query_params         = [ 'start_date', 'end_date', 's' ];
+		$this->display_report_page( $sensei_analysis_course_users, 'course_users', $exclude_query_params );
 	}
 
 	/**
 	 * An individual Lesson view for analysis, showing all the Learners
 	 *
 	 * @since  1.2.0
-	 * @param integer $lesson_id
+	 *
+	 * @param int $lesson_id
 	 * @return void
 	 */
-	public function analysis_lesson_users_view( $lesson_id ) {
-
-		// Load Analysis data
+	public function analysis_lesson_users_view( int $lesson_id ) {
 		$sensei_analysis_lesson_users = $this->load_data_object( 'Lesson', $lesson_id );
+		$exclude_query_params         = [ 's' ];
+		$this->display_report_page( $sensei_analysis_lesson_users, 'lesson_users', $exclude_query_params );
+	}
+
+	private function display_report_page( Sensei_List_Table $list_table, $nav_type = null, array $exclude_query_params = [] ) {
 		// Wrappers
 		do_action( 'analysis_before_container' );
 		do_action( 'analysis_wrapper_container', 'top' );
 
-		$this->analysis_lesson_users_nav();
+		$this->display_nav( $nav_type );
 		do_action( 'sensei_analysis_after_headers' );
 
+		$list_table->views();
 		?>
-		<div id="poststuff" class="sensei-analysis-wrap course-profile">
-			<div class="sensei-analysis-main">
-				<?php $sensei_analysis_lesson_users->display(); ?>
-			</div>
-			<div class="sensei-analysis-extra">
-				<?php do_action( 'sensei_analysis_extra' ); ?>
```

Comment:
No usages found on GH.
Consider removing or deprecating?

Reply:
It's a really old hook (8+ years). I think it would be better to deprecate it first.